lib/stringify.cpp

```diff
@@ -81,15 +81,15 @@ cstring printf_format(const char* fmt_str, ...) {
 // printf into a string
 cstring vprintf_format(const char* fmt_str, va_list ap) {
     static char buf[128];
+    va_list ap_copy;
+    va_copy(ap_copy, ap);
```

Comment:
You fail to call va_end if the string doesn't overflow.  I'm not sure if that will cause problems, but the spec does require that you call va_end exactly once for each va_start or va_copy, so it risks potential undefined behavior.

Reply:
Yeah I realized while pushing the commit, I am fixing both your comments in the next commit. Thanks for the help!